core-referrals/db/migrate/20230502211749_create_core_referrers_and_referrals.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+class CreateCoreReferrersAndReferrals < ActiveRecord::Migration[6.0]
+  def change
+    create_table :core_referrers, if_not_exists: true, id: :uuid do |t|
+      t.citext :code, null: false
+      t.integer :referrals_count, default: 0
+      t.references :referrer, polymorphic: true, type: :uuid, null: false
+
+      t.timestamps
+    end
+
+    add_index :core_referrers, :code, unique: true, if_not_exists: true
```

Comment:
Could eliminate this line in favor of augmenting the column definition with  `index: { unique: true }`